docs/maintainer.md

```diff
@@ -0,0 +1,212 @@
+This records not only "ours" and "theirs", but also the
+common ancestor. This allows the maintainer to see the
+intent of each change and aids in resolution.
+
+
+### `git submodule sync`
+
+Avoid using `git submodule sync` as it sets internal state
+that is easy to forget, and which can wreak havoc if you do
+forget it.
+
+## Applying patches and patch series
+
+Patches can arrive via mailing lists and github pull
+requests. Either way, they should be applied to the
+`next` branch, not `master`.
+
+Sometimes, patches and patch series have dependencies.
+It's important to verify that those dependencies are
+satisfied. In the case of patch series, make sure
+the series base contains the dependency, otherwise
+bisectability is compromised.
+
+A series base can be found with the command
+
+    git merge-base remote/series_branch_or_tag origin/master
+
+
+### Applying patches and patch series from the mailing list
+
+Before you begin, check out the `next` branch and pull
+from scylla.git to be up to date. If the pull isn't clean,
+abort the merge and use `git pull --rebase`. Examine the result
+to see if you forgot to push a previously applied patch.
+
+### Applying single patches from the mailing list
+
+Save the patch(es) to some directory, and use the command
+
+    git am -im3 /path/to/patches/*.eml
+
+to apply the patches. `-i` makes the process interactive and
+lets you edit the commit message, `-m` sets the Message-Id
+tag (which is used by Commit Bot to set the Reply-To header,
+so that the commit acknowledgement appears as a response to
+the patch), and `-3` enables 3-way merging which is rumored
+to reduce conflicts.
+
+Use `git push` to publish the patches.
+
+### Applying patch series from the mailing list
+
+Indentify the git url and branch/tag identifier, and issue
+the command
+
+    git pull --log --no-ff <url> <branch/tag>
+
+The `--log` flag generates a list of patches in the commit log,
+while `--no-ff` ensures a merge commit is created.
+
+Copy the cover letter subject and body to the merge commit's
+subject and body, respectively. Make sure the merge commit
+supplies enough information to understand what the series is
+doing without having to read individual commits.
+
+Use `git push` to publish the patches.
+
+### Applying patches and patch series from github pull requests
+
+A common contributor mistake is to base patches on `next`
+rather than on `master`. This results in random commits
+appearing in the pull request, if `next` is edited for some
+reason.
+
+Ensure that the target branch in the pull request page is
+set to `scylladb:next` (click Edit and change it if that's
+not the case).
+
+When merging, verify that github didn't mangle the commit
+log, check both your and the contributor's email address
+are correct (no @noreply.users.github.com or similar
+address, or home vs. work addresses).
+
+### Applying single patches from github pull requests
+
+Select "Squash and merge" and follow through.
```

Comment:
I thought this should be "Rebase and merge".
Why would you want to squash a single patch?

Reply:
That's what I do and it works for me. I didn't try the other thing. I guess it makes more sense, but need to verify github doesn't mangle the commits.